videoclip.lua

```diff
@@ -431,6 +475,92 @@ function main_menu:update()
+
+            local r = subprocess({ -- This is technically blocking, but I don't think it has any real consequences ..?
+                os_type == 'windows' and 'curl.exe' or 'curl', '-s',
+                '-F', 'reqtype=fileupload',
+                '-F', 'time=' .. config['litterbox_expire'],
+                '-F', 'fileToUpload=@"' .. outfile .. '"',
+                endpoint
+            })
+
+            -- Exit codes in the range [0, 99] are returned by cURL itself.
+            -- Any other exit code means the shell failed to execute cURL.
+            if r.status < 0 or r.status > 99 then
+                notify("Error: Failed to upload. Make sure cURL is installed and in your PATH.", "error", 3)
+                return
+            end
+            if r.status ~= 0 then
+                notify("Error: Failed to upload to " .. config.litterbox and "litterbox.catbox.moe" or "catbox.moe", "error", 2)
+                return
+            end
+
+            -- Copy to clipboard
+            if os_type == 'windows' then
+                local clipboard_command = {
+                    'powershell.exe', '-command',
+                    'Set-Clipboard -Value ' .. r.stdout
+                }
+                
+                local cb = subprocess(clipboard_command)
+                -- local status = mp.command('run ' .. clipboard_command)
+                if cb.status ~= 0 then
+                    notify("Failed to copy URL to clipboard, trying to open in browser instead. (Make sure PowerShell is installed)", "warn", 4)
+                    mp.commandv('run', open_utility, r.stdout)
+                    return
+                end
+            end
+
+            if os_type == 'macos' then
+                local clipboard_command = {
+                    "pbcopy"
+                }
+                local cb = subprocess(clipboard_command, r.stdout)
+                if cb.status ~= 0 then
+                    notify("Failed to copy URL to clipboard, trying to open in browser instead. (Make sure pbcopy is installed)", "warn", 4)
+                    mp.commandv('run', open_utility, r.stdout)
+                    return
+                end
+            end
+
+            if os_type == "linux" then
+                local session_type = io.popen('/bin/sh -c "echo $XDG_SESSION_TYPE"'):read("*a")
+
+                local cb
+                if session_type == "x11\n" then
+                    local clipboard_command = {
+                        "/bin/sh", "-c", "'echo " .. r.stdout .. " | xclip -sel clip'"
+                    }
+                    cb = subprocess(clipboard_command)
+                else
+                    -- wl-copy is from wl-clipboard
+                    local clipboard_command = {
+                        "wl-copy", r.stdout
+                    }
+                    cb = subprocess(clipboard_command)
+                end
+                if cb.status ~= 0 then
```

Comment:
After the subprocess change the alternative action will always trigger since the command returns -3